lightstep/resource_dashboard.go

```diff
@@ -30,9 +32,33 @@ func getUnifiedQuerySchema() map[string]*schema.Schema {
 	return sma
 }
 
-func getQueriesFromUnifiedDashboardResourceData(queriesIn []client.MetricQueryWithAttributes) []interface{} {
+func getQueriesFromUnifiedDashboardResourceData(
+	queriesIn []client.MetricQueryWithAttributes,
+	dashboardID string,
+	chartID string,
+) ([]interface{}, error) {
+
 	var queries []interface{}
 	for _, q := range queriesIn {
+		// Check if a query is using the legacy representation.  In that case,
+		// it can't be represented in the "query_string" field and the call should
+		// fail. In the future, we should attempt to automatically convert legacy -> query
+		// string format. At the moment, there's no public API to implement this, so
+		// at least provide a clarifying error message.
+		if q.Type != "tql" {
```

Comment:
my memory is failing me, is this always what the public api calls query types? it doesnt try to rename this as `uql` or `query_string` or anything? i can also go check but figured you might know offhand

Reply:
Per our offline, conversation, it should be "tql" at least for this version of the API (i.e. we'd need to update the provider anyway if it changed).